syntax = "proto3";

package zen.remote;

import "common.proto";

service RenderingUnitService
{
  rpc New(NewRenderingUnitRequest) returns (EmptyResponse);

  rpc Delete(DeleteResourceRequest) returns (EmptyResponse);

  rpc GlEnableVertexAttribArray(GlEnableVertexAttribArrayRequest)
      returns (EmptyResponse);

  rpc GlDisableVertexAttribArray(GlDisableVertexAttribArrayRequest)
      returns (EmptyResponse);

  rpc GlVertexAttribPointer(GlVertexAttribPointerRequest)
      returns (EmptyResponse);
}

message NewRenderingUnitRequest
{
  uint64 id = 1;
  uint64 virtual_object_id = 2;
}

message GlEnableVertexAttribArrayRequest
{
  uint64 id = 1;
  uint32 index = 2;
}

message GlDisableVertexAttribArrayRequest
{
  uint64 id = 1;
  uint32 index = 2;
}

message GlVertexAttribPointerRequest
{
  uint64 id = 1;
  uint32 index = 2;
  uint64 buffer_id = 3;
  int32 size = 4;
  uint64 type = 5;
  bool normalized = 6;
  int32 stride = 7;
  uint64 offset = 8;
}
